mewarpx/mewarpx/mwxrun.py

```diff
@@ -103,27 +103,36 @@ def init_timestep(self, V_anode, DT=None, CFL_factor=None, V_grid=5):
         return self.dt
 
 
-    def init_run(self):
+    def init_run(self, restart=False, checkpoint=""):
         if self.initialized:
             raise RuntimeError(
                 "Attempted to initialize the mwxrun class multiple times.")
+        try:
+            if restart:
+                self.simulation.amr_restart = os.path.join("diags", checkpoint)
```

Comment:
The `run_restart` function should be called here to use that functionality to find the most recent checkpoint file. So I'd say passing the `checkpoint_prefix` (which should have a standard default value for most / all of our runs) as an optional parameter to `init_run` would be the way to go. Then that `checkpoint_prefix` value is passed to `run_restart` to find which checkpoint to restart from.

Reply:
Ok makes sense